src/Data/String/Regex.purs

```diff
@@ -32,22 +35,91 @@ foreign import showRegex' :: Regex -> String
 instance showRegex :: Show Regex where
   show = showRegex'
 
--- | Flags that control matching.
-type RegexFlags =
-  { global :: Boolean
+type RegexFlagsRec =
+  { global     :: Boolean
   , ignoreCase :: Boolean
-  , multiline :: Boolean
-  , sticky :: Boolean
-  , unicode :: Boolean
+  , multiline  :: Boolean
+  , sticky     :: Boolean
+  , unicode    :: Boolean
   }
 
+-- | Flags that control matching.
+data RegexFlags = RegexFlags RegexFlagsRec
```

Comment:
Oh, never mind the `Newtype` instance. It won't work because of the row there. Ugh, we need `~` types.